design/baremetal-operator/uefi-http-boot.md

```diff
@@ -0,0 +1,126 @@
+
+## Proposal
+
+Add a new hardware driver called `redfish-uefihttp`, similar to
+`redfish-virtualmedia`, to the Baremetal Operator.
+The new hardware driver supports UEFI HTTP boot using the ironic
+`redfish-https` boot interface.
+
+## Design Details
+
+Using exclusively Redfish API, the accepted value of `Firmware Interface` and
+`RAIDInterface` for the `redfish-uefihttp` hardware driver will be `redfish`.  
+Secure Boot will also be supported.
+
+When the `redfish-uefihttp` hardware driver is used we need to configure the
+`redfish-https` boot interface in ironic, which is currently not enabled
+in ironic-image.
+
+### Implementation Details/Notes/Constraints
+
+It should be enough to add the new `redfish-uefihttp` hardware driver to
+BMO to support the UEFI HTTP boot feature.  
+As for the `redfish-virtualmedia` driver, the new `redfish-uefihttp` relies
+on network boot from an ISO image that is composed by Ironic, but unlike
+Virtual Media boot, it requires a Provisioning Network to be configured.  
+This is because the UEFI HTTP boot is essentially "secured PXE" designed
+to overcome limitations of the traditional PXE boot method, and it requires
+functional DNS, DHCP and HTTP servers to be configured.  
+The ISO image is provided by the HTTP server and the host boots from it
+exactly like it happened with a standard ramdisk during PXE boot.
+
+### Risks and Mitigations
+
+The UEFI HTTP boot provides a high level of security for a network boot
+process, relying on secure communication between BMC and HTTP server to
+transfer all the files and images needed to boot the host.
+
+### Work Items
+
+- Add a new hardware driver called `redfish-uefihttp` to BMO.
+- Add `redfish-https` to supported boot interfaces to the ironic-image.
+- Update metal3 documentation with details on the new hardware driver and
+how to use it.
+
+### Dependencies
+
+Ironic implementation of the `redfish-https` boot interface [has been
+already completed](https://review.opendev.org/c/openstack/ironic/+/900964).
+
+### Test Plan
+
+The new hardwre driver can currently be tested only on supported hardware
+as the Redfish support must be enabled and supported by the BMC.  
+Although the support for UEFI HTTP boot has been added to sushy-tools, so
+the current BMO e2e tests set can be expanded to cover this scenario.
+
+Unit tests will be implemented as usual.
+
+### Upgrade / Downgrade Strategy
+
+The new hardware driver will be available upgrading the version of Baremetal
+Operator.
+
+### Version Skew Strategy
+
+None
+
+## Drawbacks
+
+Adding one more hardware driver with a different boot interface complicates
+the decision process to decide which boot interface to use, in particular in
+this case between `redfish-virtualmedia` and `redfish-uefihttp` since they both
+rely on network booting from an ISO image.  
+In general, the main difference is that the UEFI HTTP boot supports Secure
+Boot, while virtual media does not, but on the other hand, no configuration
```

Comment:
How come no configuration drive functionality? Is it not possible to boot IPA thus not possible to send over the config drive via IPA API? or what do you men by no configuration drive functionality? Could you please clarify this?

Reply:
I don't think Riccardo meant that, but I agree, it is confusing.

Reply:
yeah, sorry for the confusion, I had to unconfuse myself too here!
this is not related to IPA at all
what I meant is that no static configuration can be injected at boot time, DHCP is a hard requirement
does that help?

Reply:
Bare with me I am bit slow to understand this, what do you mean by static configuration? 
Do you mean we can't use the feature with servers that have static IPs for their interfaces? So same way as for iPXE the UEFI HTTP booting also requires the BOOTP protocol thus there must be a DHCP server present at all times to facilitate the boot media discovery?

Reply:
@Rozzii exactly that
I will change the wording